packages/ember-glimmer/lib/component-managers/custom.ts

```diff
@@ -52,125 +139,129 @@ export interface ComponentArguments<T = {}> {
   * `update()` - invoked when the arguments passed to a component change
   * `getContext()` - returns the object that should be
 */
-export default class CustomComponentManager<T> extends AbstractComponentManager<
-  CustomComponentState<T> | null,
-  DefinitionState
-> {
-  constructor(private delegate: CustomComponentManagerDelegate<T>) {
+export default class CustomComponentManager<ComponentInstance = Opaque>
+  extends AbstractComponentManager<CustomComponentState<ComponentInstance>, DefinitionState>
+  implements
+    WithStaticLayout<
+      CustomComponentState<ComponentInstance>,
+      DefinitionState<ComponentInstance>,
+      OwnedTemplateMeta,
+      RuntimeResolver
+    > {
+  constructor(private delegate: ManagerDelegate<ComponentInstance>) {
     super();
   }
 
   create(
     _env: Environment,
     definition: DefinitionState,
     args: Arguments,
-    dynamicScope: DynamicScope
-  ): CustomComponentState<T> {
+    _dynamicScope: DynamicScope
+  ): CustomComponentState<ComponentInstance> {
     const { delegate } = this;
-    const capturedArgs = args.named.capture();
-
-    const component = delegate.create({
-      args: capturedArgs.value(),
-      ComponentClass: (definition.ComponentClass as any) as T,
-    });
+    const capturedArgs = args.capture();
 
-    const { view: parentView } = dynamicScope;
-
-    if (parentView !== null && parentView !== undefined) {
-      addChildView(parentView, component);
-    }
-
-    dynamicScope.view = component;
+    let invocationArgs = valueForCapturedArgs(capturedArgs);
+    const component = delegate.createComponent(definition.ComponentClass, invocationArgs);
 
     return new CustomComponentState(delegate, component, capturedArgs);
   }
 
-  update({ component, args }: CustomComponentState<T>) {
-    this.delegate.update(component, args.value());
+  update({ component, args }: CustomComponentState<ComponentInstance>) {
+    this.delegate.updateComponent(component, valueForCapturedArgs(args));
```

Comment:
so, for example this would be

```ts
update({ delegate, component, args }: CustomComponentState<ComponentInstance>) {
  delegate.updateComponent(component, valueForCapturedArgs(args));
}
```